File: problems/733. Flood Fill/Solution.java
class Solution {

    class IntPair {
        final int x;
        final int y;
        IntPair(int x, int y) {
            this.x = x;
            this.y = y;
        }
    }

    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        if (image[sr][sc] == color) return image;
        Stack<IntPair> stack = new Stack<>();
        stack.push(new IntPair(sr, sc));
        int initialColor = image[sr][sc];
        while (!stack.empty()) {
            IntPair elem = stack.pop();
            image[elem.x][elem.y] = color;
            if (elem.x > 0 && image[elem.x - 1][elem.y] == initialColor)
                stack.push(new IntPair(elem.x - 1, elem.y));
            if (elem.x < image.length - 1 && image[elem.x + 1][elem.y] == initialColor)
                stack.push(new IntPair(elem.x + 1, elem.y));
            if (elem.y > 0 && image[elem.x][elem.y - 1] == initialColor)
                stack.push(new IntPair(elem.x, elem.y - 1));
            if (elem.y < image[0].length - 1 && image[elem.x][elem.y + 1] == initialColor)
                stack.push(new IntPair(elem.x, elem.y + 1));
        }
        return image;
    }
}

File: problems/733. Flood Fill/Solution.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        if image[sr][sc] == color: return image
        initialColor = image[sr][sc]
        stack = [(sr, sc)]
        while len(stack) > 0:
            x, y = stack.pop()
            image[x][y] = color
            if (x > 0) and (image[x - 1][y] == initialColor):
                stack.append((x - 1, y))
            if (x < len(image) - 1) and (image[x + 1][y] == initialColor):
                stack.append((x + 1, y))
            if (y > 0) and (image[x][y - 1] == initialColor):
                stack.append((x, y - 1))
            if (y < len(image[0]) - 1) and (image[x][y + 1] == initialColor):
                stack.append((x, y + 1))
        return image
